### ISSUE TITLE
GCC 8.3.1: format overflow

### ISSUE THREAD

**Issue by DavidVujic** [yfinkelstein/node-zookeeper#154]
This error has occurred on a Linux machine. I got this from a `node-zookeeper` user:

```bash
make all-am make[2]: Entering directory '/[real-path-removed]/node_modules/zookeeper/deps/zookeeper/src/c' /bin/sh ./libtool --tag=CC --mode=compile gcc -DHAVE_CONFIG_H -I. -I./include -I./tests -I./generated -Wall -Werror -g -O2 -D_GNU_SOURCE -MT zookeeper.lo -MD -MP -MF .deps/zookeeper.Tpo -c -o zookeeper.lo `test -f 'src/zookeeper.c' || echo './'`src/zookeeper.c libtool: compile: gcc -DHAVE_CONFIG_H -I. -I./include -I./tests -I./generated -Wall -Werror -g -O2 -D_GNU_SOURCE -MT zookeeper.lo -MD -MP -MF .deps/zookeeper.Tpo -c src/zookeeper.c -fPIC -DPIC -o zookeeper.o src/zookeeper.c: In function 'format_endpoint_info': src/zookeeper.c:3471:21: error: '%d' directive writing between 1 and 5 bytes into a region of size between 0 and 127 [-Werror=format-overflow=] sprintf(buf,"%s:%d",addrstr,ntohs(port));
```

**Comment by davidvujic-svenskaspel** [comment]
Workaround:
```CFLAGS=-Wno-error=format-overflow npm install```

**Comment by DavidVujic** [comment]
Something to investigate: is this fixed in the `3.5.x` version of the ZooKeeper C Client?
